core/modules/css/KvInterfaceImplMySql.cc

```diff
@@ -0,0 +1,503 @@
+    }
+
+    unsigned int count(0);
+    if (not extractIntValueFromSqlResults(&results, &count)) {
+        throw CssError("failed to extract int value from query");
+    }
+
+    bool foundIt(false);
+    if (1 == count) {
+        foundIt = true;
+    } else if (count > 1) {
+        throw CssError("multiple keys for key");
+    }
+    transaction.commit();
+    return foundIt;
+}
+
+
+std::vector<std::string>
+KvInterfaceImplMySql::getChildren(std::string const& parentKey) {
+
+    // get the children with a /fully/qualified/path
+    KvTransaction transaction(_conn);
+    std::vector<std::string> strVec = _getChildrenFullPath(parentKey);
+    transaction.commit();
+
+    // trim off the parent key, leaving only the last item in the path.
+    for (std::vector<std::string>::iterator strItr = strVec.begin(); strItr != strVec.end(); ++strItr) {
+        size_t loc = strItr->find_last_of(KEY_PATH_DELIMITER)+1;
+        if (strItr->size() == loc) {
+            // technically this shouldn't happen because keys shoudln't end with '/', but a little safety does not hurt...
+            continue;
+        }
+        *strItr = strItr->substr(loc, strItr->size());
+    }
+    return strVec;
+}
+
+
+std::vector<std::string>
+KvInterfaceImplMySql::_getChildrenFullPath(std::string const& parentKey) {
+    // A transaction should be in process here.
+
+    unsigned int parentId;
+    if (not _getIdFromServer(parentKey, &parentId)) {
+        if (not exists(parentKey)) {
+            throw NoSuchKey(parentKey);
+        }
+    }
+
+    boost::format fmQuery("SELECT kvKey FROM kvData WHERE parentKvId='%1%'");
+    fmQuery % parentId;
+    std::string query = fmQuery.str();
+
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    LOGF(_logger, LOG_LVL_DEBUG, "_getChildrenFullPath - executing query: %1%" % query);
+    if (not _conn.runQuery(query, results, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "_getChildrenFullPath - %1% failed with err:%2%" % query % errObj.errMsg());
+        throw CssError((boost::format("_getChildrenFullPath - error:%1% from query:%2%") % errObj.errMsg() % query).str());
+    }
+
+    errObj.reset();
+    std::vector<std::string> strVec;
+    if (not results.extractFirstColumn(strVec, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "_getChildrenFullPath - failed to extract children from %1% failed with err:%2%" % query % errObj.errMsg());
+        throw CssError((boost::format("_getChildrenFullPath - failed to extract children error:%1% from query:%2%") % errObj.errMsg() % query).str());
+    }
+
+    return strVec;
+}
+
+
+
+void
+KvInterfaceImplMySql::deleteKey(std::string const& key) {
+    KvTransaction transaction(_conn);
+    _delete(key);
+    transaction.commit();
+}
+
+
+void
+KvInterfaceImplMySql::_delete(std::string const& key) {
+    // A transaction should be in process here.
+
+    // recursively delete child keys first
+    std::vector<string> childKeys = _getChildrenFullPath(key);
```

Comment:
THere should be more efficient way to delete them than just doing one-by-one recursively, but I'm OK with this for now, we can do optimizations later.